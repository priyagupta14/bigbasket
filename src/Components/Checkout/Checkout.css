.checkout-form{
  display:flex;
  flex-direction: column;
  border: 1px solid black;
  margin: 2rem;
  box-shadow: 10px 10px 5px rgb(143, 137, 137);
  padding: 1rem;
  border-radius: 25px;
}
.checkout-form>Form{
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0.5rem 4rem;

}
Form>p{
  display: flex;
  flex-direction: column;
  
}
/* p>label{
  display: flex;
}
Form>button{
  margin: 15px 10rem;
} */
@media screen and (max-width: 50rem) {
    .checkout-form>Form{
      display: flex;
      flex-direction: column;
      justify-content: center;
      padding: 0.5rem 1rem;
    
    }
  }