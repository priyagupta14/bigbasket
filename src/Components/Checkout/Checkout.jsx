/* eslint-disable react/no-access-state-in-setstate */
/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import './Checkout.css';

export default class Checkout extends Component {
  constructor() {
    super();
    this.state = {
      formValues: {
        name: '',
        address: '',
        phoneNo: '',
        email: '',
      },
      formErrors: {
        name: '',
        address: '',
        phoneNo: '',
        email: '',
      },
      formValidity: {
        name: false,
        address: false,
        phoneNo: false,
        email: false,
      },
      // isSubmitting: false,
    };
  }

  // handleChange=(event) => {
  //   if (event.target.name === 'name' &&
  //  event.target.value.length > 6) { console.log(event.target.value); }
  //   this.setState(({ [event.target.name]: event.target.value }));
  // }

  handleChange=({ target }) => {
    const { formValues } = this.state;
    formValues[target.name] = target.value;
    this.setState({ formValues });
    this.handleValidation(target);
  }

  handleValidation= (target) => {
    const { name, value } = target;
    const fieldValidationErrors = this.state.formErrors;
    const validity = this.state.formValidity;
    const isEmail = name === 'email';
    const isPhoneNo = name === 'phoneNo';
    const emailTest = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
    validity[name] = value.length > 0;
    fieldValidationErrors[name] = validity[name] ? '' : `${name} is required`;
    if (validity[name]) {
      if (isEmail) {
        validity[name] = emailTest.test(value);
        fieldValidationErrors[name] = validity[name] ? '' : `${name} should be an valid`;
      }
      if (isPhoneNo) {
        validity[name] = value.length === 10;
        fieldValidationErrors[name] = validity[name] ? '' : `${name} should be an valid`;
      }
    }
    this.setState({
      formErrors: fieldValidationErrors,
      formValidity: validity,
    });
  }

  render() {
    const { formValues, formValidity, formErrors } = this.state;
    return (
      <div className="checkout-form">
        <form>
          <p>Name:</p>
          <input type="text" name="name" onChange={this.handleChange} value={this.state.name} />
          <div className="invalid-feedback">{formErrors.name}</div>
          <p>Address:</p>
          <input type="text" name="address" onChange={this.handleChange} value={this.state.address} />
          <div className="invalid-feedback">{formErrors.address}</div>
          <p>Phone no:</p>
          <input type="number" name="phoneNo" placeholder="enter phone no" onChange={this.handleChange} value={this.state.phoneNo} />
          <div className="invalid-feedback">{formErrors.phoneNo}</div>
          <p>email:</p>
          <input type="email" name="email" placeholder="enter email" onChange={this.handleChange} value={formValues.email} />
          <div className="invalid-feedback">{formErrors.email}</div>
          <input type="submit" value="Submit" />
        </form>
      </div>

    );
  }
}
