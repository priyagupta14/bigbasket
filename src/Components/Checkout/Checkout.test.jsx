import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import useEvent from '@testing-library/user-event';

import userEvent from '@testing-library/user-event';
import Checkout from './Checkout';

describe(Checkout.name, () => {
  let mockProp;
  beforeEach(() => {
    mockProp = {
      productList: {
        abc: [
          {
            id: 1,
            name: 'abc',
            price: 20,
            inCartCount: 5,
          },
        ],
      },
      orderCart: jest.fn(),
    };
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match the snapshot', async () => {
    const { container } = render(<Checkout {...mockProp} />);
    expect(container).toMatchSnapshot();
  });

  // test();
});
