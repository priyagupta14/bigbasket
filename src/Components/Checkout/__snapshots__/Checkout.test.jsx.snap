// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout should match the snapshot 1`] = `
<div>
  <div
    class="checkout-form"
  >
    <h2>
      Enter Details...
    </h2>
    <form
      action="#"
    >
      <p>
        <label
          for="Name"
        >
            Name 
        </label>
        <input
          name="Name"
          type="text"
          value=""
        />
      </p>
      <p>
        <label
          for="Address"
        >
            Address 
        </label>
        <input
          name="Address"
          type="text"
          value=""
        />
      </p>
      <p>
        <label
          for="PhoneNo"
        >
            Phone no. 
        </label>
        <input
          name="PhoneNo"
          type="number"
          value=""
        />
      </p>
      <p>
        <label
          for="Email"
        >
            Email
        </label>
        <input
          name="Email"
          type="text"
          value=""
        />
      </p>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
